src/shogun/features/DenseFeatures.cpp

```diff
@@ -1036,24 +1036,34 @@ template<class ST> CFeatures* CDenseFeatures<ST>::create_merged_copy(
 	SGMatrix<ST> data(num_features, num_vectors_merged);
 
 	/* copy data of this instance */
-	SG_DEBUG("copying matrix of this instance\n")
-	memcpy(data.matrix, feature_matrix.matrix,
-			num_features*num_vectors*sizeof(ST));
+	SG_DEBUG("copying matrix of this instance\n");
+	if (m_subset_stack->has_subsets())
+	{
+		SGMatrix<ST> active_feats_mat=get_feature_matrix();
```

Comment:
Agreed. A separate private method should simply accept a raw ptr (or maybe a SGMatrix) to copy things into. If the ptr (SGMatrix.matrix) is empty or doesn't have enough space to memcopy things into, it reallocates.

Then this method as well as the other get_feature_matrix method can rely on that method.